src/main/java/com/algorand/algosdk/transaction/AtomicTransactionComposer.java

```diff
@@ -326,6 +376,77 @@ public ExecuteResult execute(AlgodClient client, int waitRounds) throws Exceptio
         return new ExecuteResult(txInfo.confirmedRound, this.getTxIDs(), retList);
     }
 
+    public static class SimulateResult {
+        // The result of the transaction group simulation
+        private SimulateResponse simulateResponse;
+        // For each ABI method call in the executed group (created by the AddMethodCall method), this
+        // list contains information about the method call's return value
+        private List<ABIMethodResult> methodResults;
+
+        // Getter and setter for simulateResponse
+        public SimulateResponse getSimulateResponse() {
+            return simulateResponse;
+        }
+
+        public void setSimulateResponse(SimulateResponse simulateResponse) {
+            this.simulateResponse = simulateResponse;
+        }
+
+        // Getter and setter for methodResults
+        public List<ABIMethodResult> getMethodResults() {
+            return methodResults;
+        }
+
+        public void setMethodResults(List<ABIMethodResult> methodResults) {
+            this.methodResults = methodResults;
+        }
+    }
+
+    /**
+     * Parses a single ABI Method transaction log into a ABI result object.
+     *
+     * @param method
+     * @param methodResult
+     * @param pendingTransactionResponse
+     * @return An ABIMethodResult object
+     */
+    public ABIMethodResult parseMethodResponse(Method method, ABIMethodResult methodResult, PendingTransactionResponse pendingTransactionResponse) {
+        ABIMethodResult returnedResult = methodResult;
+        try {
+            returnedResult.setTransactionInfo(pendingTransactionResponse);
+            if (!method.returns.type.equals("void")) {
+                List<byte[]> logs = pendingTransactionResponse.logs;
+                if (logs == null || logs.isEmpty()) {
+                    throw new Exception("App call transaction did not log a return value");
+                }
+
+                byte[] lastLog = logs.get(logs.size() - 1);
+                if (lastLog.length < 4 || !hasPrefix(lastLog, ABI_RET_HASH)) {
+                    throw new Exception("App call transaction did not log a return value");
+                }
+
+                returnedResult.setRawReturnValue(Arrays.copyOfRange(lastLog, 4, lastLog.length));
```

Comment:
minor: I would prefer referencing the return hash length instead of hardcoding 4
```suggestion
                returnedResult.setRawReturnValue(Arrays.copyOfRange(lastLog, ABI_RET_HASH.length, lastLog.length));
```